Objective-C/CBLDatabase.mm

```diff
@@ -950,102 +952,138 @@ - (BOOL) saveDocument: (CBLDocument*)document
+                                        remoteDoc: remoteDoc error: &err];
         
-        // Save resolved document:
-        if (![self saveResolvedDocument: resolvedDoc
-                           withLocalDoc: localDoc
-                              remoteDoc: remoteDoc
-                                  error: outError])
-            return false;
+        if ($equal(err.domain, CBLErrorDomain) && err.code == CBLErrorConflict)
+            continue;
         
-        return t.commit() || convertError(t.error(), outError);
+        if (outError)
+            *outError = err;
+        
+        return success;
     }
 }
 
-
-- (CBLDocument*) resolveConflictBetweenLocalDoc: (CBLDocument*)localDoc
-                                   andRemoteDoc: (CBLDocument*)remoteDoc {
-    if (remoteDoc.isDeleted)
-        return remoteDoc;
-    else if (localDoc.isDeleted)
-        return localDoc;
-    else if (localDoc.generation > remoteDoc.generation)
-        return localDoc;
-    else if (localDoc.generation < remoteDoc.generation)
-        return remoteDoc;
-    else if ([localDoc.revID compare: remoteDoc.revID] > 0)
-        return localDoc;
-    else
-        return remoteDoc;
-}
-
-
 - (BOOL) saveResolvedDocument: (CBLDocument*)resolvedDoc
                  withLocalDoc: (CBLDocument*)localDoc
                     remoteDoc: (CBLDocument*)remoteDoc
                         error: (NSError**)outError
 {
-    if (resolvedDoc != localDoc)
-        resolvedDoc.database = self;
-    
-    // The remote branch has to win, so that the doc revision history matches the server's.
-    CBLStringBytes winningRevID = remoteDoc.revID;
-    CBLStringBytes losingRevID = localDoc.revID;
-    
-    alloc_slice mergedBody;
-    C4RevisionFlags mergedFlags = 0;
-    if (resolvedDoc != remoteDoc) {
-        // Unless the remote revision is being used as-is, we need a new revision:
-        mergedBody = [resolvedDoc encode: outError];
-        if (!mergedBody)
-            return false;
-        if (resolvedDoc.isDeleted)
-            mergedFlags |= kRevDeleted;
-    }
-    
-    // Tell LiteCore to do the resolution:
-    C4Document *rawDoc = localDoc.c4Doc.rawDoc;
-    C4Error c4err;
-    if (!c4doc_resolveConflict(rawDoc,
-                               winningRevID,
-                               losingRevID,
-                               mergedBody,
-                               mergedFlags,
-                               &c4err)
-        || !c4doc_save(rawDoc, 0, &c4err)) {
-        return convertError(c4err, outError);
+    CBL_LOCK(self) {
+        C4Transaction t(_c4db);
+        if (!t.begin())
+            return convertError(t.error(), outError);
+        
+        if (!resolvedDoc) {
+            if (!localDoc && remoteDoc)
+                resolvedDoc = remoteDoc; // local deleted, keep remote
+            
+            if (!remoteDoc && localDoc)
+                resolvedDoc = localDoc; // remote deleted, keep local
+        }
```

Comment:
localDoc and remoteDoc will never be nil here. I think the logic should be something like below (You may need to clean it up):

```
if (!resolvedDoc) {
   if (localDoc.isDeleted)
       resolved = localDoc
   else if (remoteDoc.isDeleted)
      resolved = remoteDoc
}
```

Reply:
we are assigning nil, if the localDoc or remoteDoc is deleted in the above function right? line 986, 987.

Reply:
Yes